modules/js-and-css/speculation-rules/hooks.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Hook callbacks used for Speculation Rules.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Prints the speculation rules in a cross-browser compatible way.
+ *
+ * For browsers that do not support speculation rules yet, the rules will not be loaded.
+ *
+ * @since n.e.x.t
+ */
+function plsr_print_speculation_rules() {
+	$rules = plsr_get_speculation_rules();
+	if ( empty( $rules ) ) {
+		return;
+	}
+
+	$script = <<<JS
+( function() {
+	if ( ! HTMLScriptElement.supports || ! HTMLScriptElement.supports( 'speculationrules' ) ) {
+		console.log( 'speculation rules not supported' );
+		return;
+	}
+
+	var specScript = document.createElement( 'script' );
```

Comment:
Nit: Might as well use ES6

```suggestion
	const specScript = document.createElement( 'script' );
```

Reply:
@westonruter Is it safe to use ES6 here? I know it's widely supported at this point, but as far as I can tell this code otherwise works with older browsers as well. If that is indeed the case, then using `const` over `var` is not enough benefit IMO to make the code _less_ compatible.

Reply:
Since there will be zero browsers that support Speculation Rules API that don't also support ES6 (as it is only new browsers that support it), it seems fine to not require ES5.  But this is moot if https://github.com/WordPress/performance/pull/733#discussion_r1203016080 is implemented.

Reply:
Right, but the reason for this script is specifically the case where browsers _don't_ support Speculation Rules API, hence my point :)

Reply:
Nevertheless, I think those specific browsers are so few and far between that we shouldn't worry about them. I just checked and usage of all versions of IE adds up to 0.47%, which is well below the 1% threshold for [WordPress browser support](https://make.wordpress.org/core/handbook/best-practices/browser-support/). In fact, IE11 hasn't been supported [since WP 5.8](https://make.wordpress.org/core/2021/04/22/ie-11-support-phase-out-plan/). I've been inquiring about this in relation to optimizing emoji-loader which would be facilitated by not having to support IE11. I raised this as a [question in #core Slack](https://wordpress.slack.com/archives/C02RQBWTW/p1686243321666769).

In any case, [`const` would be supported by IE11 here](https://caniuse.com/const). But you're right, `var` works equivalently here.